File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<int[]> q = new LinkedList<>();
        
        for(int i=0; i<priorities.length; i++){
            // System.out.println(Arrays.toString(new int[]{priorities[i], i}));
            q.add(new int[]{priorities[i], i});
        }
        
        int max = Arrays.stream(priorities).max().getAsInt();
        
        while(true){
            max = Arrays.stream(priorities).max().getAsInt();
            int[] tmp = q.poll();
            
            if(tmp[0] == max){
                answer += 1;
                int idx = Arrays.stream(priorities)
                    .boxed()
                    .collect(Collectors.toList())
                    .indexOf(max);
                priorities[idx] = -1;
                if(tmp[1] == location){
                    break;
                }
            }else{
                q.add(tmp);
            }
        }
        
        // System.out.println(q);
        
        // System.out.println(idx);
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    arr = deque()
    for idx, priority in enumerate(priorities):
        arr.append((priority, idx))

    priorities.sort(reverse = True)
    
    while True:
        priority, idx = arr.popleft()
        
        if idx == location:
            if priority == priorities[answer]:
                break;
            arr.append((priority, idx))
        else :
            if priority == priorities[answer]:
                answer += 1
            else :
                arr.append((priority, idx))
        
    return answer+1